src/app/qgsvectorlayerproperties.cpp

```diff
@@ -249,6 +249,12 @@ QgsVectorLayerProperties::QgsVectorLayerProperties(
   diagLayout->addWidget( diagramPropertiesDialog );
   mDiagramFrame->setLayout( diagLayout );
 
+  // WMS Name as layer short name
+  mLayerShortNameLineEdit->setText( layer->shortName() );
+  // WMS Name validator
+  QRegExp shortNameRx( "^[A-Za-z][A-Za-z0-9\._-]*" );
```

Comment:
No, I would search for an ancestor, if any. Or you could also create a new class for the validator and reuse that class in all places.